### DISCUSSION TITLE
Memory used

### DISCUSSION THREAD

**Discussion by cesarcastmore** [CloudBytes-Academy/CloudBytes.dev#141] [General]
Hello! Excellent contribution regarding Docker. 

I just have one question. When I was uploading my image to Lambda, I noticed that it required a lot of memory, and I think that could significantly increase costs. What are the differences between layers and Docker in Lambda?

I was following this documentation https://dev.to/awscommunity-asean/creating-an-api-that-runs-selenium-via-aws-lambda-3ck3 
 and managed to do it for version 3.7, but I was unsuccessful with version 3.9. Some of the comments below suggest using Docker, but I realized that using Docker requires increasing the Lambda's memory. Do you know of any alternatives to using Docker that won't consume a lot of memory and increase costs

**Comment by rehanhaider** [comment]
> What are the differences between layers and Docker in Lambda?

Not a lot of difference under the hood, as they both use containers. In case of Layers, the Dockerfile & entrypoint is pre-defined, while the other overrides that. 

So you can always [customise the template.yaml](https://cloudbytes.dev/snippets/run-selenium-in-aws-lambda-for-ui-testing#templateyaml) to reduce the memory being used. 

What is the memory allocation you are targeting?
